pkg/gatherers/conditional/conditional_gatherer.go

```diff
@@ -237,17 +253,21 @@ func (g *Gatherer) updateAlertsCacheFromClient(ctx context.Context, metricsClien
 			klog.Info(logPrefix + "metric is nil")
 			continue
 		}
-
+		alert := Alert{
+			Labels: make(map[string]string),
+		}
 		for _, label := range metric.GetLabel() {
 			if label == nil {
 				klog.Info(logPrefix + "label is nil")
 				continue
 			}
-
 			if label.GetName() == "alertname" {
-				g.firingAlerts[label.GetValue()] = true
+				alert.Name = label.GetValue()
+				continue
 			}
+			alert.Labels[label.GetName()] = label.GetValue()
 		}
+		g.firingAlerts[alert.Name] = append(g.firingAlerts[alert.Name], alert)
```

Comment:
Is there any reason to have the `Name` as an explicit parameter of the `Alert` type when the only way to access it is by its name in the `firingAlerts` map? You can also get the name from the `Labels` and this way there's an extra level of a type nested into the map that doesn't really seem very necessary to me, as `Name` is the only parameter is has beside the label map. I thought you would just create a new type alias `AlertLabels` -> `map[string]string` and that `firingAlerts` would be `map[string][]AlertLabels`.

Reply:
Looking at the rest of the code, I don't see `Name` being actually used anywhere in a significant way, right? Couldn't the `Alert` type be simplified to just a string:string map to only store the labels, then?

Reply:
I guess no special reason :) OK let me update it